src/views/Dashboard.vue

```diff
@@ -0,0 +1,120 @@
+<!--
+  - @copyright Copyright (c) 2020 Julius Härtl <jus@bitgrid.net>
+  -
+  - @author Julius Härtl <jus@bitgrid.net>
+  -
+  - @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -
+  -->
+
+<template>
+	<DashboardWidget :items="importantMessages"
+		:show-more-url="''"
+		:loading="loading"
+		@hide="() => {}"
+		@markDone="() => {}">
+		<template v-slot:default="{ item }">
+			<DashboardWidgetItem :item="getWidgetItem(item)">
+				<template v-slot:avatar>
+					<Avatar v-if="item.from" :email="item.from[0].email" :display-name="item.from[0].label" />
+				</template>
+			</DashboardWidgetItem>
+		</template>
+		<template v-slot:empty-content>
+			<div class="empty-content">
+				<img class="empty-content__image" :src="emptyImage">
+				<p class="empty-content__text">
+					{{ t('mail', 'No messages found yet') }}
+				</p>
+				<a v-if="accounts.length === 2" :href="accountSetupUrl" class="button">{{ t('mail', 'Set up an account') }}</a>
+			</div>
+		</template>
+	</DashboardWidget>
+</template>
+
+<script>
+import { fetchEnvelopes } from '../service/MessageService'
+import { loadState } from '@nextcloud/initial-state'
+import { generateUrl, imagePath } from '@nextcloud/router'
+import Avatar from '../components/Avatar'
+import { DashboardWidget, DashboardWidgetItem } from '@nextcloud/vue-dashboard'
+
+const accounts = loadState('mail', 'mail-accounts')
+
+export default {
+	name: 'Dashboard',
+	components: {
+		Avatar,
+		DashboardWidget,
+		DashboardWidgetItem,
+	},
+	data() {
+		return {
+			messages: null,
+			accounts,
+			fetchedAccounts: 0,
+			emptyImage: imagePath('mail', 'newsletter.svg'),
+			accountSetupUrl: generateUrl('/apps/mail/#/setup'),
+		}
+	},
+	computed: {
+		loading() {
+			return this.fetchedAccounts < this.accounts.length
+		},
+		importantMessages() {
+			return this.messages ? this.messages.sort((a, b) => b.dateInt - a.dateInt).slice(0, 7) : []
```

Comment:
eslint is right: `Array.sort` modifies the original array, we should copy the array or use `sort` from lodash